wlauto/workloads/googleslides/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,469 @@
+        repeatClickView(decreaseFont, 20);
+        getUiDevice().pressBack();
+
+        // get image from gallery and insert
+        insertSlide("Title Only");
+        clickView(BY_DESC, "Insert");
+        clickView(BY_TEXT, "Image", true);
+        clickView(BY_TEXT, "Recent");
+        try {
+            UiObject image = new UiObject(new UiSelector().resourceId("com.android.documentsui:id/date").instance(2));
+            image.clickAndWaitForNewWindow();
+        } catch (UiObjectNotFoundException e) {
+            clickView(BY_ID, "com.android.documentsui:id/date", true);
+        }
+
+        // last slide
+        insertSlide("Title Slide");
+        // insert "?" shape
+        clickView(BY_DESC, "Insert");
+        clickView(BY_TEXT, "Shape");
+        clickView(BY_TEXT, "Buttons");
+        clickView(BY_DESC, "actionButtonHelp");
+        UiObject resize = getViewByDesc("Bottom-left resize");
+        UiObject shape = getViewByDesc("actionButtonHelp");
+        UiObject subtitle = getViewByDesc("subTitle");
+        resize.dragTo(subtitle, 40);
+        shape.dragTo(subtitle, 40);
+        enterTextInSlide("title", "THE END. QUESTIONS?");
+
+        sleep(1);
+        getUiDevice().pressBack();
+        deleteDocument(docName);
+    }
+
+    public void insertSlide(String slideLayout) throws Exception {
+        sleep(1); // a bit of time to see previous slide
+        UiObject view = getViewByDesc("Insert slide");
+        view.clickAndWaitForNewWindow();
+        view = getViewByText(slideLayout);
+        view.clickAndWaitForNewWindow();
+    }
+
+    public void enterTextInSlide(String viewName, String textToEnter) throws Exception {
+        UiObject view = getViewByDesc(viewName);
+        view.click();
+        view.setText(textToEnter);
+        try {
+            clickView(BY_DESC, "Done");
+        } catch (UiObjectNotFoundException e) {
+            clickView(BY_ID, "android:id/action_mode_close_button");
+        }
+        SystemClock.sleep(200);
+    }
+
+    public void saveDocument(String docName) throws Exception {
+        timer = new Timer();
+        timer.start();
+        clickView(BY_TEXT, "SAVE");
+        clickView(BY_TEXT, "Device");
+        timer.end();
+        results.put("save_dialog1", timer);
+        // Allow SD card access if requested
+        UiObject permissionView = new UiObject(new UiSelector().textContains("Allow Slides"));
+        if (permissionView.waitForExists(DIALOG_WAIT_TIME_MS)) {
+            clickView(BY_TEXT, "Allow");
+        }
+
+        timer = new Timer();
+        timer.start();
+        UiObject filename = getViewById(PACKAGE_ID + "file_name_edit_text");
+        filename.clearTextField();
+        filename.setText(docName);
+        clickView(BY_TEXT, "Save", CLASS_BUTTON);
+        timer.end();
+        results.put("save_dialog2", timer);
+        // Overwrite if prompted
+        UiObject overwriteView = new UiObject(new UiSelector().textContains("already exists"));
```

Comment:
Should not happen under normal circumstances. But when the test sometimes failed for whatever reason, the created file would not be deleted and thus be on the device during the next run.

Note that part of the test is to delete the file in-app, so it isn't removed as part workload teardown. One scenario where I've seen this happen is when long-click on the document list fails.

It is not included in timings, so shouldn't skew results.